src/btree/node.rs

```diff
@@ -64,16 +64,23 @@ impl Node {
 		log: &mut LogWriter,
 	) -> Result<(Option<(Separator, Child)>, bool)> {
 		loop {
-			if changes.len() > 1 {
-				if changes[0].key() == changes[1].key() &&
-					!changes[0].is_reference_ops() &&
-					!changes[1].is_reference_ops()
-				{
+			if changes.len() > 1 && !changes[0].is_reference_ops() {
+				let mut skip = false;
+				for next in changes[1..].iter() {
+					if changes[0].key() == next.key() {
+						if !next.is_reference_ops() {
+							skip = true;
+							break
+						}
+					} else {
+						break
```

Comment:
Shouldn't we `continue` here instead of `break` because it is fine to skip the operation if there is an operation that overrides the current one it is fine if it is not consecutive? Or is there some bad interactions with reference ops?

Reply:
I love github, I am not sure if line 76 or 73, for 76 it means we are on a different key so we check no other Set operation coming up (keys are sorted), for 73 we did see a Set operation afterward so we can skip this current set operation and we don't need to look further.
I mean the idea is for changes at the same key, all Set before the last one can be skip. Reference (does rc + 1) or dereference (rc - 1) cannot as we need the sum of these calls.
(and doing the sum at this level is not doable because we don't want to fetch the current rc and after rc reach 0 any following reference operation are noops (unless there was a Set in between)). Actually we could safely merge consecutive Reference and even Dereference as long as there is no chance to reach 0, but would need to change some Reference enum to contains a counter (which would not be bad).

Reply:
> If break at line 76 it means we are on a different key so we check no other Set operation coming up (keys are sorted),

Ok! Thank you! I missed that the key were sorted.